sympy/core/symbol.py

```diff
@@ -96,10 +96,10 @@ def _symbol(s, matching_symbol=None, **assumptions):
         raise ValueError('symbol must be string for symbol name or Symbol')
 
 
-def _uniquely_named_symbol(xname, exprs=(), compare=str, modify=None, **assumptions):
+def uniquely_named_symbol(xname, exprs=(), compare=str, modify=None, **assumptions):
```

Comment:
Why are we renaming it?

Reply:
@smichr suggested that we'd promote it to public. Please refer to #19066.